python/tskit/tables.py

```diff
@@ -3020,6 +3026,99 @@ def union(
                 record=json.dumps(provenance.get_provenance_dict(parameters))
             )
 
+    def time_join(self, recent, time, continue_nodes=None):
```

Comment:
Since we need the trees, I think we should require that the input argument is a tree sequence, so we call it ``recent_ts``.